Add Roc Curve and Precision/Recall Curve to the metrics

GabrielePicco opened the pull request:
Description:
Add Roc Curve and Precision/Recall Curve to the metrics

Check list:
* [x] New tests are added (if a new feature is added)
* [x] New doc strings: description and/or example code are in RST format

vfdev-5 commented on ignite/contrib/metrics/precision_recall_curve.py:
```diff
@@ -0,0 +1,41 @@
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+
+    ROC_AUC expects y to be comprised of 0's and 1's. y_pred must either be probability estimates or confidence
```
```suggestion
    PrecisionRecallCurve expects y to be comprised of 0's and 1's. y_pred must either be probability estimates or confidence
```

vfdev-5 commented on ignite/contrib/metrics/roc_curve.py:
```diff
@@ -0,0 +1,41 @@
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+
+    ROC_AUC expects y to be comprised of 0's and 1's. y_pred must either be probability estimates or confidence
```
```suggestion
    RocCurve expects y to be comprised of 0's and 1's. y_pred must either be probability estimates or confidence
```

GabrielePicco commented:
Thanks @vfdev-5 for the review! I added the metrics to the module and fixed the typos.
Having them in two separate files seems clearer to me, but if you think it's better I can join them.

vfdev-5 commented:
> but if you think it's better I can join them.
@GabrielePicco yes, please ;)

GabrielePicco commented:
Done @vfdev-5
